runtime/src/bank.rs

```diff
@@ -7510,23 +7525,29 @@ impl Bank {
             // As a workaround for
             // https://github.com/solana-labs/solana-program-library/issues/374, ensure that the
             // spl-token 2 native mint account is owned by the spl-token 2 program.
+            let old_account_data_size;
             let store = if let Some(existing_native_mint_account) =
                 self.get_account_with_fixed_root(&inline_spl_token::native_mint::id())
             {
+                old_account_data_size = existing_native_mint_account.data().len();
                 if existing_native_mint_account.owner() == &solana_sdk::system_program::id() {
                     native_mint_account.set_lamports(existing_native_mint_account.lamports());
                     true
                 } else {
                     false
                 }
             } else {
+                old_account_data_size = 0;
                 self.capitalization
                     .fetch_add(native_mint_account.lamports(), Relaxed);
                 true
             };
 
             if store {
                 self.store_account(&inline_spl_token::native_mint::id(), &native_mint_account);
+                let data_size_delta = (native_mint_account.data().len() as i64)
+                    .saturating_sub(old_account_data_size as i64);
+                self.update_accounts_data_size_delta_off_chain(data_size_delta);
```

Comment:
seems useful to have an update function that takes (new_lamports, prev_lamports) and does the casts, saturating subs, delta update, etc.

Reply:
I think I was thinking combine both of these:
``` 
let data_size_delta =
                    compute_data_size_delta(old_account.data().len(), new_account.data().len());
                self.update_accounts_data_size_delta_off_chain(data_size_delta);
                ```

Reply:
`self.update_accounts_data_size_difference_off_chain(old, new);`?

Reply:
name is bad. just the idea.

Reply:
oh, I am so much happy